### ISSUE TITLE
Issue with lights in aframe from blender models

### ISSUE THREAD

**Issue by Michael-Pascale** [aframevr/aframe#4432]
I was making a model in blender for testing in aframe, using all default settings in blender. When I used it in aframe, I noticed that it did not appear to be using the light resources. I confirmed that all the normals were facing the right direction, and that no gltf export flags were different than what should be there (Not using unlit settings). 
In a separate occasion with a model from elsewhere (an obj file), I used blender to export into a glb and it used the light properly, but the original obj did not. (Not including those files, just mentioning).
I am not sure if this is a bug or if I am doing something wrong, but all of the models view correctly in blender, paint3d and 3dviewer on windows 10.

- A-Frame Version: 1.0.2
- Platform / Device: Windows 10

[error.zip](https://github.com/aframevr/aframe/files/4144809/error.zip)

**Comment by donmccurdy** [comment]
Could you describe what you mean by "it did not appear to be using the light resources"? What do you expect to see, and what are you actually seeing? Do you mean for the file to contain a light itself (it doesn't currently), or do you have other lights in the A-Frame scene? Is the model entirely black, or just darker than you expect?

**Comment by Michael-Pascale** [comment]
![image](https://user-images.githubusercontent.com/49280454/74040157-28b11180-4991-11ea-91a0-a0f00dccac63.png)

----------------------------------------
This is with a default scene in a frame. The left is a simple obj file with an mtl. The right is that same file exported from blender as a glb. I no longer have the demo I had before, but when I made a glb within blender and loaded it into aframe, it looked like the one on the left. I did check the material settings to make sure they matched aframes specs, and the model in the picture is the only one that has successfully worked.

**Comment by donmccurdy** [comment]
> The left is a simple obj file with an mtl.

> when I made a glb within blender and loaded it into aframe, it looked like the one on the left.

It sounds like you're saying the OBJ and glTF look the same?

Sorry, I'm not sure what you're asking here. We need enough information to reproduce the problem, to help. Consider loading the model into a viewer like https://gltf-viewer.donmccurdy.com/ — if it looks like the Blender original there, but not in A-Frame, it's probably a matter of getting the lighting right in A-Frame. If it looks wrong there, too, it might be another issue.

**Comment by Michael-Pascale** [comment]
![image](https://user-images.githubusercontent.com/49280454/74051781-a16e9880-49a6-11ea-971e-fd4f17494257.png)
----------------------------------------
Above is as this particular model looks in blender. I made the model there.
----------------------------------------
![image](https://user-images.githubusercontent.com/49280454/74052090-4a1cf800-49a7-11ea-8305-a08f40757ea2.png)
----------------------------------------
Above is the exported glb in aframe. I made no changes to the default scene, which uses a directional light if I am not mistaken.
----------------------------------------
![image](https://user-images.githubusercontent.com/49280454/74052201-80f30e00-49a7-11ea-9000-30ccf37179c9.png)
This is what the model looks like in your model viewer.
---------------------------------------
![image](https://user-images.githubusercontent.com/49280454/74052330-a8e27180-49a7-11ea-9322-fc61df041f51.png)
---------------------------------------
This is the validation report that your viewer sent me. 
--------------------------------------
![image](https://user-images.githubusercontent.com/49280454/74052456-e3e4a500-49a7-11ea-9159-d67d5e02f241.png)
--------------------------------------
I turned the environment lighitng off, and the result looks like this, which is the same as what the default scene in aframe shows.
--------------------------------------
![image](https://user-images.githubusercontent.com/49280454/74052559-20180580-49a8-11ea-9a45-6c72714274e0.png)
-----------------------------------------
Lastly, the view from the bottom of the model in your viewer. If you need anything else, let me know and I will provide it. I am not sure why it is acting like this. If your environment light is the same as an ambient light value, then the results are that it looks incorrect in both your viewer and aframe.

**Comment by donmccurdy** [comment]
Thanks, that's helpful. Enabling `renderer.colorManagement` might be the place to start: see https://aframe.io/docs/1.0.0/components/renderer.html.

**Comment by donmccurdy** [comment]
> If your environment light is the same as an ambient light value

These are quite different – ambient lights won't affect metallic materials. Since the model doesn't appear metallic, you might want to make sure you've set metalness to `0` in Blender. Anything metallic requires an environment map, and it takes a bit of work to add those in A-Frame.

**Comment by Michael-Pascale** [comment]
I turned down all of those settings in blender and did another export. I also changed the settings to match those in the link you sent. Below is what the results were:
-------------------------------------
![image](https://user-images.githubusercontent.com/49280454/74203633-4a82f080-4c3e-11ea-9378-8d73537d14d4.png)
-------------------------------------
After messing with some settings, turning off physicallycorrectlights yields:
-------------------------------------
![image](https://user-images.githubusercontent.com/49280454/74203668-6ab2af80-4c3e-11ea-96c5-8d9f148bfd2c.png)
-------------------------------------
This is closer to what I was expecting the object to look like. It looks like if I use these settings I can get the objects to be lit up, albeit they wont get to use any pbr if I understand this. What could this mean?

**Comment by donmccurdy** [comment]
>  It looks like if I use these settings I can get the objects to be lit up, albeit they wont get to use any pbr if I understand this...

I don't know exactly what settings you changed, but disabling `physicallyCorrectLights` is not equivalent to disabling PBR. `physicallyCorrectLights=true` simply means that for lights with a distance property (point, spot) the distance attenuation follows inverse square law (i.e. physically-based falloff). Practically, this usually makes things a little darker for the same light intensity value. 

PBR refers to much more than just how far the lights shine.

**Comment by netgfx** [comment]
I think I have the same or similar issue, I'm loading this glb file 
https://cdn.glitch.com/9565c650-fdb4-4b2f-a1fc-f2e8506c0092%2Fkey.glb?v=1586977089656
However it looks like this (versus the google model viewer) 
![image (6)](https://user-images.githubusercontent.com/864248/79377628-a7639580-7f64-11ea-8ffe-c43fc6bfc19f.png)
The "orange" one is from A-Frame.

Code is like this:
```
<a-scene device-orientation-permission-ui="enabled: false" background="transparent: true"
            cursor="rayOrigin:mouse" >
<a-asset-item id="object" src="./assets/treasure/key.glb"></a-asset-item>

<a-entity gltf-model="#object" position="0 1 -4" scale="1 1 1" rotation="0 0 0" animation="property: rotation; to: 0 360 0; from:0 0 0; easing:linear; loop: true; dur: 10000"  renderer="antialias: true;
                   colorManagement: true;
                   physicallyCorrectLights: true;"></a-entity>

    <a-camera mouse-cursor>
            <a-cursor fuse="true" color="yellow"></a-cursor>
     </a-camera>
</a-scene>

**Comment by donmccurdy** [comment]
The `renderer="colorManagement: true"` property would need to be on the scene, it is a scene-wide setting.

**Comment by netgfx** [comment]
Yes it did improve a lot, however still not close to the model-viewer, I wonder what settings does the model-viewer use to show that true golden color, maybe something to do with light?
![image (7)](https://user-images.githubusercontent.com/864248/79431478-30b1b100-7fd3-11ea-921d-dea97e51feae.png)

**Comment by donmccurdy** [comment]
Visible color would depend on lighting – grabbing the texture from Blender (below) I don't think A-Frame is wrong. But I'm not sure what model-viewer is doing differently here, it may be the choice of a particular environment map for lighting.

<img width="300" src="https://user-images.githubusercontent.com/1848368/79524382-3af49f00-8015-11ea-8b8e-31baa2009dde.png" alt="key texture golden yellow">

**Comment by netgfx** [comment]
I see, thanks for the information! I'll try some different settings. I also see (on a different model) that on the model-viewer the object is reflecting more so maybe they are not using an ambient light, or maybe some different more glossy material?

**Comment by donmccurdy** [comment]
They definitely use IBL (image-based lighting or environment maps), and might also use some ambient or other lights on top of that. But there are multiple ways to do IBL, and a lot depends on the texture itself, so I'm not sure how to reproduce that without inspecting their code.

**Comment by olivierFalloff** [comment]
Hello
I have been using a-frame since 2018 to show my scenographies to the directors I work for (thank you for your fantastic work). Since aframe-v1.1.0.min I have been forced to use the GLTF format but I'm not happy with the lights. Objects seem to emit light like emit. The Gltf format is less beautiful in equivalent light than the collada format. The GLTF format shines, it is heavier and it
there is too much aliaising. I cannot find the lighting I had with the collada format. Please give me a way to find my lights with the collada format ...I tried: "collada-model-legacy.js" with "ColladaLoader.js" in another folder but it doesn't work. Do you have a solution?

**Comment by donmccurdy** [comment]
See https://github.com/aframevr/aframe/issues/3793#issuecomment-779480924.

**Comment by olivierFalloff** [comment]
Thanks for your reply, I took your advice and found the answer (see the link below). This will allow me to smoothly transition to the promising GLTF format. Thank yoyi again for your VR work. It is a real tool for scenographers and architects ...

https://stackoverflow.com/questions/66196396/how-to-use-a-collada-model-with-aframe-v1-1-0-min
